cmd/snap-seccomp/main.go

```diff
@@ -0,0 +1,619 @@
+		return fmt.Errorf("too many arguments specified for syscall '%s' in line %q", tokens[0], line)
+	}
+
+	// fish out syscall
+	secSyscall, err := seccomp.GetSyscallFromName(tokens[0])
+	if err != nil {
+		// FIXME: use structed error in libseccomp-golang when
+		//   https://github.com/seccomp/libseccomp-golang/pull/26
+		// gets merged. For now, ignore
+		// unknown syscalls
+		return nil
+	}
+
+	var conds []seccomp.ScmpCondition
+	for pos, arg := range tokens[1:] {
+		var cmpOp seccomp.ScmpCompareOp
+		var value uint64
+		var err error
+
+		if arg == "-" { // skip arg
+			continue
+		}
+
+		if strings.HasPrefix(arg, ">=") {
+			cmpOp = seccomp.CompareGreaterEqual
+			value, err = readNumber(arg[2:])
+		} else if strings.HasPrefix(arg, "<=") {
+			cmpOp = seccomp.CompareLessOrEqual
+			value, err = readNumber(arg[2:])
+		} else if strings.HasPrefix(arg, "!") {
+			cmpOp = seccomp.CompareNotEqual
+			value, err = readNumber(arg[1:])
+		} else if strings.HasPrefix(arg, "<") {
+			cmpOp = seccomp.CompareLess
+			value, err = readNumber(arg[1:])
+		} else if strings.HasPrefix(arg, ">") {
+			cmpOp = seccomp.CompareGreater
+			value, err = readNumber(arg[1:])
+		} else if strings.HasPrefix(arg, "|") {
+			cmpOp = seccomp.CompareMaskedEqual
+			value, err = readNumber(arg[1:])
+		} else {
+			cmpOp = seccomp.CompareEqual
+			value, err = readNumber(arg)
+		}
+		if err != nil {
+			return fmt.Errorf("cannot parse token %q (line %q)", arg, line)
+		}
+
+		var scmpCond seccomp.ScmpCondition
+		if cmpOp == seccomp.CompareMaskedEqual {
+			scmpCond, err = seccomp.MakeCondition(uint(pos), cmpOp, value, value)
+		} else {
+			scmpCond, err = seccomp.MakeCondition(uint(pos), cmpOp, value)
+		}
+		if err != nil {
+			return fmt.Errorf("cannot parse line %q: %s", line, err)
+		}
+		conds = append(conds, scmpCond)
+	}
+
+	// Default to adding a precise match if possible. Otherwise
+	// let seccomp figure out the architecture specifics.
+	if err = secFilter.AddRuleConditionalExact(secSyscall, seccomp.ActAllow, conds); err != nil {
+		err = secFilter.AddRuleConditional(secSyscall, seccomp.ActAllow, conds)
+	}
+
+	return err
+}
+
+// For architectures that support a compat architecture, when the
+// kernel and userspace match, add the compat arch, otherwise add
+// the kernel arch to support the kernel's arch (eg, 64bit kernels with
+// 32bit userspace).
+func addSecondaryArches(secFilter *seccomp.ScmpFilter) error {
+	// note that all architecture strings are in the dpkg
+	// architecture notation
+	var compatArch string
+
+	// common case: kernel and userspace have the same arch. We
+	// add a compat architecture for some architectures that
+	// support it, e.g. on amd64 kernel and userland, we add
+	// compat i386 syscalls.
+	if arch.UbuntuArchitecture() == arch.UbuntuKernelArchitecture() {
+		switch arch.UbuntuArchitecture() {
+		case "amd64":
+			compatArch = "i386"
+		case "arm64":
+			compatArch = "armhf"
+		case "ppc64el":
+			compatArch = "powerpc"
+		}
```

Comment:
`powerpc` is only valid with `ppc64` because they are both bid endian, not `ppc64el` which is little endian, but see my comments above where you omit `ppc64` as a valid arch.